operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ConfigParameterParser.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.operator.resource;
+
+
+import io.strimzi.operator.common.InvalidConfigurationException;
+import io.strimzi.operator.common.model.Labels;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Abstraction for things which convert a single configuration parameter value from a String to some specific type.
+ */
+public interface ConfigParameterParser<T> {
+
+    /**
+     * Parses the string based on its type
+     *
+     * @param configValue config value in String format
+     * @throws InvalidConfigurationException if the given configuration value is not supported
+     * @return the value based on its type
+     */
+    T parse(String configValue) throws InvalidConfigurationException;
+
+    /**
+     * A java string
+     */
+
+    ConfigParameterParser<String> STRING = configValue -> configValue;
+
+    /**
+     * A non empty java string
+     */
+    ConfigParameterParser<String> NON_EMPTY_STRING = configValue -> {
+        if (configValue == null || configValue.isEmpty()) {
+            throw new InvalidConfigurationException("Failed to parse. Value cannot be empty or null");
+        } else {
+            return configValue;
+        }
+    };
+
+    /**
+     * A semicolon-delimited list of strings.
```

Comment:
What do we use semicolon-delimited lists for? I know we use comma-delimited lists, but not aware of semicolons.

Reply:
We are using it for `MAINTENENCE_TIME_WINDOW in the `UserOperatorConfig` class

Reply:
Changing it to comma seperated lists was giving error

Reply:
That is fine, if it simecolon sepaarted it is semicolon separated. But you should not name it just `LIST`.